lib/WP_Auth0_LoginManager.php

```diff
@@ -573,6 +562,31 @@ protected function die_on_login( $msg = '', $code = 0 ) {
 		wp_die( $html );
 	}
 
+	/**
+	 * @param $id_token
+	 * @return object
+	 * @throws WP_Auth0_InvalidIdTokenException
+	 */
+	private function decode_id_token( $id_token ) {
+		$idTokenIss  = 'https://' . $this->a0_options->get( 'domain' ) . '/';
+		$sigVerifier = null;
+		if ( 'RS256' === $this->a0_options->get( 'client_signing_algorithm' ) ) {
+			$jwks        = ( new WP_Auth0_JwksFetcher() )->getKeys();
+			$sigVerifier = new WP_Auth0_AsymmetricVerifier( $jwks );
+		} elseif ( 'HS256' === $this->a0_options->get( 'client_signing_algorithm' ) ) {
+			$sigVerifier = new WP_Auth0_SymmetricVerifier( $this->a0_options->get( 'client_secret' ) );
+		}
+
+		$verifierOptions = [
+			'nonce'   => WP_Auth0_Nonce_Handler::get_instance()->get_once(),
```

Comment:
was about to say "typo" but I'm not sure if this is intentional or not. On other places, you call `get_unique`. Please, evaluate.

Reply:
`get_unique` gets the unique, stored value and `get_once()` gets the value and deletes it.

Reply:
on platforms like Java, taking the Map class for example, that would be called [get](https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#get(java.lang.Object)) and [remove](https://docs.oracle.com/javase/7/docs/api/java/util/Map.html#remove(java.lang.Object)) (which also returns the existing value). Unless this is a pattern for PHP, or it's already documented the expected behavior of each method, consider renaming them. Not a blocker

Reply:
The classes and methods here, this one in particular, are unlikely to be used by a developer so I'm not too worried about them. `get_once()` feels clear to me for what it's doing. `get_unique()` could probably change (should probably just be `get` since there is only ever one value for the class).